### PR TITLE
Adding match command to omegascript

### PR DESCRIPTION
**Pull request by richhiey1996**
I have added a match function to transform.cc and hooked it up in query.cc.
Do check this patch out and let me know if there is I can improve or anything that I have gone wrong with. :)

### CODE REVIEW

File: xapian-applications/omega/query.cc [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * Copyright 2001 James Aylett
  * Copyright 2001,2002 Ananova Ltd
  * Copyright 2002 Intercede 1749 Ltd
- * Copyright 2002,2003,2004,2005,2006,2007,2008,2009,2010,2011,2013,2014 Olly Betts
+ * Copyright 2002,2003,2004,2005,2006,2007,2008,2009,2010,2011,2013,2014,2015,2016 Olly Betts
```
Comment (jaylett) [inline xapian-applications/omega/query.cc]: Why is this change here? Olly's last change to this file was 6f15477, in 2014. Did you mean to add a line for yourself, with a 2016 date?

File: xapian-applications/omega/query.cc [hunk 2]
```diff
@@ -256,6 +257,12 @@ read_qp_flags(const string & opt_pfx, unsigned f)
 		    break;
 		}
 		break;
+	    case 'c':
+		if (strcmp(s, "cjk_ngram") == 0) {
```
Comment (jaylett) [inline xapian-applications/omega/query.cc]: Does this have anything to do with `$match{}`?

**Comment by jaylett** [comment]
I'm going to stop reviewing it at this point, because I think something strange is going on and that this commit contains other changes for some reason. Can you rebase just your `$match{}` changes onto master? (Probably easiest to reapply the bits you care about by hand then add them in using `git add -p`.) Then it should all be smaller and a lot easier to review; I'd expect the entire patch to be around +50 lines, not +400-150.

**Comment by richhiey1996** [comment]
I'm sorry this has many unwanted additions and deletions. I happened to make various additions and deletions because of working on the source tarball and trying to add the same changes to the master.

I will work on it and send another pull request. Really sorry about this.

**Comment by jaylett** [comment]
@richhiey1996 don't worry about it. Now you at least know why it's easier to work from git rather than a tarball :-)

(The `master` branch in your repository is a long way behind the upstream master, so before you tidy this up you should `git fetch upstream` or similar depending on what you called the remote to our repo. It'll make life a lot easier.)

**Comment by jaylett** [comment]
Closing in favour of #95.
